src/widget/form/settings/userinterfaceform.cpp

```diff
@@ -0,0 +1,311 @@
+        bodyUI->themeColorCBox->addItem(color);
+
+    bodyUI->themeColorCBox->setCurrentIndex(s.getThemeColor());
+    bodyUI->emoticonSize->setValue(s.getEmojiFontPointSize());
+
+    QStringList timestamps;
+    for (QString timestamp : timeFormats)
+        timestamps << QString("%1 - %2").arg(timestamp, QTime::currentTime().toString(timestamp));
+
+    bodyUI->timestamp->addItems(timestamps);
+
+    QLocale ql;
+    QStringList datestamps;
+    dateFormats.append(ql.dateFormat());
+    dateFormats.append(ql.dateFormat(QLocale::LongFormat));
+    dateFormats.removeDuplicates();
+    timeFormats.append(ql.timeFormat());
+    timeFormats.append(ql.timeFormat(QLocale::LongFormat));
+    timeFormats.removeDuplicates();
+
+    for (QString datestamp : dateFormats)
+        datestamps << QString("%1 - %2").arg(datestamp, QDate::currentDate().toString(datestamp));
+
+    bodyUI->dateFormats->addItems(datestamps);
+    bodyUI->timestamp->setCurrentText(QString("%1 - %2").arg(s.getTimestampFormat(), QTime::currentTime().toString(s.getTimestampFormat())));
+    bodyUI->dateFormats->setCurrentText(QString("%1 - %2").arg(s.getDateFormat(), QDate::currentDate().toString(s.getDateFormat())));
+
+    // Chat
+    connect(bodyUI->textStyleComboBox, &QComboBox::currentTextChanged, this, &UserInterfaceForm::onStyleUpdated);
+
+    connect(bodyUI->showWindow, &QCheckBox::stateChanged, this, &UserInterfaceForm::onShowWindowChanged);
+    connect(bodyUI->showInFront, &QCheckBox::stateChanged, this, &UserInterfaceForm::onSetShowInFront);
+
+    connect(bodyUI->groupAlwaysNotify, &QCheckBox::stateChanged, this, &UserInterfaceForm::onSetGroupAlwaysNotify);
+    connect(bodyUI->cbGroupchatPosition, &QCheckBox::stateChanged, this, &UserInterfaceForm::onGroupchatPositionChanged);
+    connect(bodyUI->cbCompactLayout, &QCheckBox::stateChanged, this, &UserInterfaceForm::onCompactLayout);
+
+    connect(bodyUI->cbDontGroupWindows, &QCheckBox::stateChanged, this, &UserInterfaceForm::onDontGroupWindowsChanged);
+    connect(bodyUI->cbSeparateWindow, &QCheckBox::stateChanged, this, &UserInterfaceForm::onSeparateWindowChanged);
+
+    // Theme
+    void (QComboBox::* currentIndexChanged)(int) = &QComboBox::currentIndexChanged;
+    connect(bodyUI->useEmoticons, &QCheckBox::stateChanged, this, &UserInterfaceForm::onUseEmoticonsChange);
+    connect(bodyUI->smileyPackBrowser, currentIndexChanged, this, &UserInterfaceForm::onSmileyBrowserIndexChanged);
+    connect(bodyUI->styleBrowser, &QComboBox::currentTextChanged, this, &UserInterfaceForm::onStyleSelected);
+    connect(bodyUI->themeColorCBox, currentIndexChanged, this, &UserInterfaceForm::onThemeColorChanged);
+    connect(bodyUI->emoticonSize, &QSpinBox::editingFinished, this, &UserInterfaceForm::onEmoticonSizeChanged);
+    connect(bodyUI->timestamp, currentIndexChanged, this, &UserInterfaceForm::onTimestampSelected);
+    connect(bodyUI->dateFormats, currentIndexChanged, this, &UserInterfaceForm::onDateFormatSelected);
```

Comment:
Same here: All UI connections can be auto-connections. Just right-click on the element and chose "show signals" from the menu. This also avoids ambiguity of method signatures like the one above.